Agent does not accept documents with mime type of application/json

### System Info

I'm just using the llama-stack starter distribution with an OpenAI inference back-end.

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### 🐛 Describe the bug

When attempting to use the API client.agents.turn.create in Typescript passing documents with a mime_type of `application/json` the document is rejected as an invalid mime type due to this line I believe:

[https://github.com/llamastack/llama-stack/blob/4a59961a6cd764db739aefeba06601dfaee68d88/llama_stack/providers/inline/agents/meta_reference/agent_instance.py#L923](https://github.com/llamastack/llama-stack/blob/4a59961a6cd764db739aefeba06601dfaee68d88/llama_stack/providers/inline/agents/meta_reference/agent_instance.py#L923)

Passing the document in as `text/json` works fine however the mime type `text/json` is not AFAIK the standard mime type for JSON.

### Error logs

```
19:19:16.720 [ERROR] Error in sse_generator
Traceback (most recent call last):
File "/usr/local/lib/python3.12/site-packages/llama_stack/distribution/server/server.py", line 184, in sse_generator
async for item in event_gen:
File "/usr/local/lib/python3.12/site-packages/llama_stack/providers/inline/agents/meta_reference/agents.py", line 181, in _create_agent_turn_streaming
async for event in agent.create_and_execute_turn(request):
File "/usr/local/lib/python3.12/site-packages/llama_stack/providers/inline/agents/meta_reference/agent_instance.py", line 189, in create_and_execute_turn
async for chunk in self._run_turn(request, turn_id):
File "/usr/local/lib/python3.12/site-packages/llama_stack/providers/inline/agents/meta_reference/agent_instance.py", line 274, in _run_turn
async for chunk in self.run(
File "/usr/local/lib/python3.12/site-packages/llama_stack/providers/inline/agents/meta_reference/agent_instance.py", line 347, in run
async for res in self._run(
File "/usr/local/lib/python3.12/site-packages/llama_stack/providers/inline/agents/meta_reference/agent_instance.py", line 460, in _run
raw_document_text = await get_raw_document_text(document)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/usr/local/lib/python3.12/site-packages/llama_stack/providers/inline/agents/meta_reference/agent_instance.py", line 915, in get_raw_document_text
raise ValueError(f"Unexpected document mime type: {document.mime_type}")
ValueError: Unexpected document mime type: application/json
```

### Expected behavior

It should accept documents with mime type of `application/json`

--- a/llama_stack/providers/inline/agents/meta_reference/agent_instance.py
+++ b/llama_stack/providers/inline/agents/meta_reference/agent_instance.py
@@ -920,7 +920,7 @@ async def get_raw_document_text(document: Document) -> str:
             DeprecationWarning,
             stacklevel=2,
         )
-    elif not (document.mime_type.startswith("text/") or document.mime_type == "application/yaml"):
+    elif not (document.mime_type.startswith("text/") or document.mime_type in ("application/yaml", "application/json")):
         raise ValueError(f"Unexpected document mime type: {document.mime_type}")
 
     if isinstance(document.content, URL):

--- a/tests/unit/providers/agent/test_get_raw_document_text.py
+++ b/tests/unit/providers/agent/test_get_raw_document_text.py
@@ -107,14 +107,48 @@ async def test_get_raw_document_text_deprecated_text_yaml_with_text_content_item
         assert "text/yaml" in str(w[0].message)
 
 
+async def test_get_raw_document_text_supports_json_mime_type():
+    """Test that the function accepts application/json mime type."""
+    json_content = '{"name": "test", "version": "1.0", "items": ["item1", "item2"]}'
+
+    document = Document(content=json_content, mime_type="application/json")
+
+    result = await get_raw_document_text(document)
+    assert result == json_content
+
+
+async def test_get_raw_document_text_with_json_url():
+    """Test that the function handles JSON URLs correctly."""
+    json_content = '{"key": "value", "list": ["item1", "item2"]}'
+
+    with patch("llama_stack.providers.inline.agents.meta_reference.agent_instance.load_data_from_url") as mock_load:
+        mock_load.return_value = json_content
+
+        document = Document(content=URL(uri="https://example.com/data.json"), mime_type="application/json")
+
+        result = await get_raw_document_text(document)
+        assert result == json_content
+        mock_load.assert_called_once_with("https://example.com/data.json")
+
+
+async def test_get_raw_document_text_with_json_text_content_item():
+    """Test that the function handles JSON TextContentItem correctly."""
+    json_content = '{"key": "value", "nested": {"array": [1, 2, 3]}}'
+
+    document = Document(content=TextContentItem(text=json_content), mime_type="application/json")
+
+    result = await get_raw_document_text(document)
+    assert result == json_content
+
+
 async def test_get_raw_document_text_rejects_unsupported_mime_types():
     """Test that the function rejects unsupported mime types."""
     document = Document(
         content="Some content",
-        mime_type="application/json",  # Not supported
+        mime_type="application/pdf",  # Not supported
     )
 
-    with pytest.raises(ValueError, match="Unexpected document mime type: application/json"):
+    with pytest.raises(ValueError, match="Unexpected document mime type: application/pdf"):
         await get_raw_document_text(document)
 
 

--- a/tests/unit/providers/agent/test_get_raw_document_text.py
+++ b/tests/unit/providers/agent/test_get_raw_document_text.py
@@ -117,20 +117,6 @@ async def test_get_raw_document_text_supports_json_mime_type():
     assert result == json_content
 
 
-async def test_get_raw_document_text_with_json_url():
-    """Test that the function handles JSON URLs correctly."""
-    json_content = '{"key": "value", "list": ["item1", "item2"]}'
-
-    with patch("llama_stack.providers.inline.agents.meta_reference.agent_instance.load_data_from_url") as mock_load:
-        mock_load.return_value = json_content
-
-        document = Document(content=URL(uri="https://example.com/data.json"), mime_type="application/json")
-
-        result = await get_raw_document_text(document)
-        assert result == json_content
-        mock_load.assert_called_once_with("https://example.com/data.json")
-
-
 async def test_get_raw_document_text_with_json_text_content_item():
     """Test that the function handles JSON TextContentItem correctly."""
     json_content = '{"key": "value", "nested": {"array": [1, 2, 3]}}'

